### ISSUE TITLE
scan has unexpected results on error

### ISSUE THREAD

**Issue by c-dante** [rpominov/kefir#148]
Possible bug / break in expectations:
```
var p = Kefir.pool();
var scan = p.scan(function(acc, x)
{
console.log(acc, x);
acc.push(x);
return acc;
}, ['DEFAULT']).log('scan');

// Error first
p.plug(Kefir.stream(function(emitter){ emitter.error(['ERROR']); emitter.end(); }));
p.plug(Kefir.sequentially(0, [1, 2, 3, 4, 5]));

// Expectation is ['DEFAULT', 1, 2, 3, 4, 5]
// Actual is very unexpected behavior (keeps emitting 5, slow emit, etc.)
```

The line in particular:
https://github.com/rpominov/kefir/blob/master/src/one-source/scan.js#L23

Seems the `emitValue(x)` should be inside the if guard, and other events should be passed on properly.

Also with this example, there seems to be a large delay in plugging a sequence onto a stream from the chrome console using the Kefir on `https://rpominov.github.io/kefir`.

**Comment by rpominov** [comment]
Yeah, how scan works with errors seems totally broken now. Thanks for the catch. I'll look into it.

**Comment by rpominov** [comment]
Fixed in `2.8.2`
